### ISSUE TITLE
Add option for rebasing repos_url repos against master

### ISSUE THREAD

**Issue by dhood** [ros2/ci#17]
If we specify a particular branch on the `ros2` repos to test for the CI jobs, [we will rebase that branch](https://github.com/ros2/ci/blob/c93a05abac0219b47ccb866c4d8f2dc9a131d462/ros2_batch_job/__main__.py#L414) against master.

There are cases when it would be useful to do the same when testing a given `ros2.repos` file, such as checking that PRs from external contributors will work correctly even after being rebased.

(There are probably also cases when we would not want to rebase)

**Comment by dirk-thomas** [comment]
The rebase always happens against the specified branch in the repos file. In the case where the repos file specifies a forked repo there is simply no knowledge about the upstream repo. So I don't see how this would be possible.

**Comment by dhood** [comment]
Couldn't we rebase the repo+branch in the custom `ros2.repos` (e.g. dhood/* + feature) against the repo+branch in the default `ros2.repos` (i.e. ros2/* + master)?

That is, mimic the process that happens when we accept these PRs.

**Comment by dirk-thomas** [comment]
Currently there is only a single repo file - not two. So yes, in theory that is possible. But that will require a significant effort in implementing the merging of two separate repo files (with all the corner cases, e.g. some repo might only be in one of the two files).

**Comment by dhood** [comment]
yep, good point about the edge cases. I am thinking about this for an improved process for testing external contributors' PRs (so more likely a 1:1 mapping). But perhaps one day PR testing will not require someone generating a custom repos file in the first place, so I think the more generic name for this ticket is good.
